src/Claim.ts

```diff
@@ -21,56 +81,26 @@ export const getClaimId = async (claim: Claim): Promise<string> => {
     .toString('hex')
 }
 
-export const getClaimSignature = async (claim: Claim, privateKey: string): Promise<string> => {
-  if (!claim.publicKey) throw new IllegalArgumentException('Cannot sign a claim that has an empty .publicKey field.')
-  if (new bitcore.PrivateKey(privateKey).publicKey.toString() !== claim.publicKey)
-    throw new IllegalArgumentException(
-      "Cannot sign this claim with the provided privateKey. It doesn\t match the claim's public key."
-    )
-  const generatedClaimId = await getClaimId(claim)
-  if (!claim.id) throw new IllegalArgumentException('Cannot sign a claim that has an empty .id field.')
-  if (claim.id !== generatedClaimId)
-    throw new IllegalArgumentException('Cannot sign a claim whose id has been altered or generated incorrectly.')
-
-  const signature = bitcore.crypto.ECDSA.sign(Buffer.from(claim.id, 'hex'), new bitcore.PrivateKey(privateKey))
-  return signature.toString()
-}
-
-export const isValidSignature = (claim: Claim): boolean => {
-  try {
-    return bitcore.crypto.ECDSA.verify(
-      Buffer.from(claim.id, 'hex'),
-      bitcore.crypto.Signature.fromString(claim.signature),
-      new bitcore.PublicKey(claim.publicKey)
-    )
-  } catch (exception) {
-    return false
-  }
-}
-
-export const createClaim = async (privateKey: string, type: ClaimType, attributes: ClaimAttributes): Promise<Claim> => {
+export const createClaim = async (
+  issuer: any,
+  type: ClaimType,
+  claimAttributes: ClaimAttributes,
+  context: ClaimContext = { '@context': {} }
+): Promise<Claim> => {
   const claim: Claim = {
-    id: '',
-    publicKey: new bitcore.PrivateKey(privateKey).publicKey.toString(),
-    signature: '',
+    '@context': { ...DefaultClaimContext['@context'], ...claimTypeDefaults[type]['@context'], ...context['@context'] },
     type,
-    created: new Date().toISOString(),
-    attributes,
+    issuer: issuer.id,
+    issuanceDate: new Date().toISOString(),
+    claim: { ...claimAttributes },
   }
   const id = await getClaimId(claim)
-  const signature = await getClaimSignature(
-    {
-      ...claim,
-      id,
-    },
-    privateKey
-  )
-  return {
+  const sign = signClaim({ nonce: cuid(), ...issuer.signingOptions })
+  return await sign({
     ...claim,
     id,
-    signature,
-  }
+  }).catch(returnError)
```

Comment:
I don't think we should catch this and return the error, we should let it throw the error and let the user of the function catch the error.